graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -69,17 +69,99 @@ def _get_last_edge_direction_and_name_to_location(location):
-        child_name_from_edge = edge_element.base_out_connection
-    return parent_name_from_edge, child_name_from_edge
+        parent_base_classname = edge_element.base_in_connection
+        child_base_classname = edge_element.base_out_connection
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
+                             u'Found: edge {} with direction {}'.format(edge_name, edge_direction))
+    return parent_base_classname, child_base_classname
+
+
+def _get_outbound_and_inbound_names_from_locations(query_metadata, child_location, parent_location):
+    """Get the out and in classnames of a location and its parent from the last edge information."""
+    edge_direction, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    parent_name_from_location = query_metadata.get_location_info(parent_location).type.name
+    child_name_from_location = query_metadata.get_location_info(child_location).type.name
+    if edge_direction == INBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = child_name_from_location
+        inbound_vertex_name = parent_name_from_location
+    elif edge_direction == OUTBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = parent_name_from_location
+        inbound_vertex_name = child_name_from_location
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
+                             u'Found: edge {} with direction {}'.format(edge_name, edge_direction))
+    return outbound_vertex_name, inbound_vertex_name
+
+
+def _probe_statistics_for_edge_count_between_vertex_pair(
+    statistics, query_metadata, child_location, parent_location
+):
+    """Probe statistics for the count of edges connecting two vertex classes."""
+    _, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    outbound_vertex_name, inbound_vertex_name = _get_outbound_and_inbound_names_from_locations(
+        query_metadata, child_location, parent_location
+    )
+    probe_result = statistics.get_edge_count_between_vertex_pair(
+        outbound_vertex_name, edge_name, inbound_vertex_name
+    )
+    return probe_result
+
+
+def _estimate_edge_count_between_vertex_pair_using_class_count(
+    schema_graph, statistics, query_metadata, child_location, parent_location
+):
+    """Estimate the count of edges connecting two vertex classes via the class_count statistic.
+
+    Given a parent location of class A and a child location of class B, this function estimates the
+    number of AB edges using class counts. If A and B are subclasses of the edge's endpoint classes
+    which we'll name C and D respectively, we only have statistics for CD edges, so in general, we
+    estimate using the assumption that CD edges are distributed among all C and D vertices
+    independently of the vertex's class.  Using this assumption, in general we estimate the
```

Comment:
"independently of the vertex's class." -> "independent of whether or not the vertex of class C is also a member of class A, and likewise for D and B."

Alternatively, split up the whole sentence since it's a bit of a run-on:

If A and B are subclasses of the edge's endpoint classes (which we'll name C and D respectively), we only have statistics for CD edges. To make estimates, we assume that CD edges are distributed among all C and D vertices independent of whether or not the vertex of class C is also a member of class A, and likewise for D and B. 

Can also add this sentence if you want to clarify.

In other words, we assume whether there is an edge from a vertex of class C to a vertex of class D is independent of whether or not the vertex of class C is also of class A and likewise for D and B.